### ISSUE TITLE
Creating a model using the modern `class` syntax

### ISSUE THREAD

**Issue by demux** [neumino/thinky#532]
I've added a method called `createModelFromClass` to my [branch](https://github.com/demux/thinky/commit/48e83e09ffe47fcc8e2c16061c747a97eb4ae7dc)

This does not change the inner workings of `thinky` but rather takes a `class` and constructs a model from it.

If there is interest, I can create some unittests, write documentation and submit a pull request. Let me know.

**Comment by neumino** [comment]
I'm not super familiar with this syntax. Is there a benefit beside the different syntax?

I'm a bit wary of adding it as I'm not super familiar with this syntax and that it's more work to maintain.

**Comment by demux** [comment]
The syntax has a lot to offer, but one major benefit, in my opinion, is getters and setters:

```javascript
@createModelFromClass
class User {
  ...
  get name() {
    return [this.firstName, this.lastName].filter((x) => x && x.length).join(' ')
  }

  set name(val) {
    const [firstName, lastName].split(' ')
    this.firstName = firstName || null
    this.lastName = lastName || null
  }
}

const user = new User({
  firstName: 'Fluffy',
  lastName: 'Bunny'
})

myModel.name
// -> Fluffy Bunny
```

You could, of course, also add `defineProperty` and `defineStaticProperty`.

But the main reason I created this is that this cleaner syntax is just much easier to comprehend for me.

I avoided making changes to existing code in order to make this easier to integrate and maintain.
If you're still concerned about the maintenance, I'd be open to creating another project for this, but I'd still need you to export the actual `Thinky` class rather than just a function that creates a new instance of `Thinky`. And I need the `_docInit` event or something to that effect.

**Comment by WaldoJeffers** [comment]
@demux It's an interesting feature, and I have been thinking about the same thing, an ORM that creates models based on pure ES6 classes. But how would you handle type definitions ?  From my understanding, Thinky require each field to have a type, in order to perform validations.

**Comment by demux** [comment]
@WaldoJeffers Yeah, just define the `schema` as a static variable on the class. I've already  implemented this on my forked version of thinky. It just needs unittesting.

**Comment by WaldoJeffers** [comment]
Ah ! I see. I'll give it a try

**Comment by demux** [comment]
If the last commit doesn't work, you can use the one before it. I'm using
that one in production.

**Comment by yoshiya0503** [comment]
decorator syntax is slightly brand new feature adopted at es7 (like python syntax).
However, I think that new syntax of es6,7 should be used initiatively.
